bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/s3/S3Unpacker.scala

```diff
@@ -40,6 +45,17 @@ class S3Unpacker()(implicit s3Client: AmazonS3) extends Unpacker {
           if exc.getMessage.startsWith("Access Denied") =>
         Some(s"Access denied while trying to read s3://$srcLocation")
 
+      case UnpackerStorageError(StoreReadError(exc: AmazonS3Exception))
```

Comment:
As below - we should think about how to do this in a way that is not specific to cloud provider.

Reply:
Let’s talk about it when I’m in this afternoon, because I was hoping it was already that way!

In line with the way we've done the scala-storage classes, there's a base `Unpacker` trait, and then implementations of that with different backends (memory, S3). I think it's okay for implementations to include cloud provider-specific messages.

If the unpacker supports multiple backends, there'll need to be a way for a user to say “unpack this bag from S3” or “unpack this bag from Azure” – so knowing the unpacker looked in a different provider might be useful!